File: OpenCV.Algorithmic.Solution/chapter-02/2.5/RGB.py
# -*- coding: utf-8 -*-
import sys
import cv2
import numpy as np

# 主函数
if __name__ == "__main__":
    # 输入图像矩阵，转换为array
    if len(sys.argv) > 1:
        image = cv2.imread(sys.argv[1], cv2.IMREAD_COLOR)
    else:
        print("Usage: python RGB.py imageFile")
        sys.exit(1)

    # 显示彩色图像
    cv2.imshow("bgr", image)

    # 得到三个颜色通道
    b = image[:,:,0]
    g = image[:,:,1]
    r = image[:,:,2]

    # 显示三个颜色通道
    cv2.imshow("b", b)
    cv2.imshow("g", g)
    cv2.imshow("r", r)

    cv2.waitKey(0)
    cv2.destroyAllWindows()

File: OpenCV.Algorithmic.Solution/chapter-02/2.5/RGB.cpp
#include <vector>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;

int main(int argc, char *argv[])
{
    // 输入图像矩阵
    Mat img = imread(argv[1], IMREAD_COLOR);
    if (img.empty())
        return -1;

    // 显示彩色图
    imshow("BGR", img);

    // 分离通道
    vector<Mat> planes;
    split(img, planes);

    // 显示B通道
    imshow("B", planes[0]);

    // 显示G通道
    imshow("G", planes[1]);

    // 显示R通道
    imshow("R", planes[2]);

    waitKey(0);

    return 0;
}
